ingenialink/ethercat/network.py

```diff
@@ -57,11 +57,15 @@ def run(self) -> None:
             for servo in self.__network.servos:
                 slave_id = servo.slave_id
                 servo_state = self.__network._get_servo_state(slave_id)
-                slave_is_alive = servo.slave.state == pysoem.NONE_STATE
-                if servo_state == NET_STATE.CONNECTED and slave_is_alive:
+                is_servo_alive = not servo.slave.state == pysoem.NONE_STATE
+                if not is_servo_alive and servo_state == NET_STATE.CONNECTED:
                     self.__network._notify_status(slave_id, NET_DEV_EVT.REMOVED)
                     self.__network._set_servo_state(slave_id, NET_STATE.DISCONNECTED)
-                if servo_state == NET_STATE.DISCONNECTED and not slave_is_alive:
+                if (
+                    is_servo_alive
+                    and servo_state == NET_STATE.DISCONNECTED
+                    and self.__network._recover_from_disconnection()
```

Comment:
Maybe I'm not used to lazy conditional evaluation of python and it took me a while to understand that the recover_from_disconnection was not being executed every loop